src/sentry/static/sentry/app/components/charts/chartZoom.jsx

```diff
@@ -0,0 +1,310 @@
+    // URL parameters are changed
+    callIfFunction(this.props.onZoom, {
+      period,
+      start: startFormatted,
+      end: endFormatted,
+    });
+
+    this.zooming = () => {
+      updateParams(
+        {
+          period,
+          start: startFormatted,
+          end: endFormatted,
+          zoom: '1',
+        },
+        this.props.router
+      );
+
+      this.saveCurrentPeriod({period, start, end});
+    };
+  };
+
+  /**
+   * Enable zoom immediately instead of having to toggle to zoom
+   */
+  handleChartReady = chart => {
+    chart.dispatchAction({
+      type: 'takeGlobalCursor',
+      key: 'dataZoomSelect',
+      dataZoomSelectActive: true,
+    });
+
+    callIfFunction(this.props.onChartReady, chart);
+  };
+
+  /**
+   * Restores the chart to initial viewport/zoom level
+   *
+   * Updates URL state to reflect initial params
+   */
+  handleZoomRestore = (evt, chart) => {
+    if (!this.history.length) {
+      return;
+    }
+
+    this.setPeriod(this.history[0]);
+
+    // reset history
+    this.history = [];
+
+    callIfFunction(this.props.onRestore, evt, chart);
+  };
+
+  handleDataZoom = (evt, chart) => {
+    const model = chart.getModel();
+    const {xAxis, series} = model.option;
+    const axis = xAxis[0];
+    const [firstSeries] = series;
+
+    // if `rangeStart` and `rangeEnd` are null, then we are going back
+    if (axis.rangeStart === null && axis.rangeEnd === null) {
+      const previousPeriod = this.history.pop();
+
+      if (!previousPeriod) {
+        return;
+      }
+
+      this.setPeriod(previousPeriod);
+    } else {
+      const start = moment.utc(firstSeries.data[axis.rangeStart][0]);
+
+      // Add a day so we go until the end of the day (e.g. next day at midnight)
+      const end = moment.utc(firstSeries.data[axis.rangeEnd][0]);
+
+      this.setPeriod({period: null, start, end}, true);
+    }
+
+    callIfFunction(this.props.onDataZoom, evt, chart);
+  };
+
+  /**
+   * Chart event when *any* rendering+animation finishes
+   *
+   * `this.zooming` acts as a callback function so that
+   * we can let the native zoom animation on the chart complete
+   * before we update URL state and re-render
+   */
+  handleChartFinished = () => {
+    if (typeof this.zooming === 'function') {
+      this.zooming();
+      this.zooming = null;
+    }
+    callIfFunction(this.props.onFinished);
+  };
+
+  render() {
+    const {
+      utc,
+      xAxis,
+      disabled,
+      children,
+
+      onZoom, // eslint-disable-line no-unused-vars
+      onRestore, // eslint-disable-line no-unused-vars
+      onChartReady, // eslint-disable-line no-unused-vars
+      onDataZoom, // eslint-disable-line no-unused-vars
+      onFinished, // eslint-disable-line no-unused-vars
+      ...props
+    } = this.props;
+
+    if (disabled) {
+      return children(props);
+    }
+
+    const hasShortInterval = useShortInterval(this.props);
+
+    let interval = hasShortInterval ? '5m' : '30m';
```

Comment:
Instead of `hasShortInterval` you could have a `getInterval` function that returns the interval instead of a boolean stating whether it's "short". I think it's a little bit nicer to keep these intervals in the one place there next to the "short" definition. It would make it easier to add more granular intervals later rather than restricting to 2.